staging/src/k8s.io/component-base/logs/api/v1/registry.go

```diff
@@ -39,14 +39,27 @@ type logFormat struct {
 	feature featuregate.Feature
 }
 
+// RuntimeControl provides operations that aren't available through the normal
+// Logger or LogSink API.
+type RuntimeControl struct {
+	// Flush ensures that all in-memory data is written.
+	// May be nil.
+	Flush func()
+
+	// SetVerbosityLevel changes the level for all Logger instances
+	// derived from the initial one. May be nil.
+	//
+	// The parameter is intentionally a plain uint32 instead of
+	// VerbosityLevel to enable implementations that don't need to import
+	// the API (helps avoid circular dependencies).
+	SetVerbosityLevel func(v uint32) error
+}
+
 // LogFormatFactory provides support for a certain additional,
 // non-default log format.
 type LogFormatFactory interface {
 	// Create returns a logger with the requested configuration.
-	// Returning a flush function for the logger is optional.
-	// If provided, the caller must ensure that it is called
-	// periodically (if desired) and at program exit.
-	Create(c LoggingConfiguration) (log logr.Logger, flush func())
+	Create(c LoggingConfiguration) (logr.Logger, RuntimeControl)
```

Comment:
It's unfortunate that Go API changes inside k8s.io/component-base/logs are needed, but the effect is limited to code which provides its own logging format, which might be none at all outside of k/k.

The "v1" refers to the user-facing API (configuration file, command line flags) and that doesn't change.

I looked (again) at whether some of this code could be moved out of the "v1" package, but that quickly fails due to circular dependencies.